functions/go/set-image/transformer/images_transformer.go

```diff
@@ -0,0 +1,170 @@
+package transformer
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/set-image/custom"
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/set-image/image_util"
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+)
+
+// SetImage supports the set-image workflow, it uses Config to parse functionConfig, Transform to change the image
+type SetImage struct {
+	// Image is the desired image
+	Image image_util.Image `json:"image,omitempty" yaml:"image,omitempty"`
+	// ConfigMap keeps the data field that holds image information
+	ConfigMap map[string]string `json:"data,omitempty" yaml:"data,omitempty"`
+	// ONLY for kustomize, AdditionalImageFields is the user supplied fieldspec
+	AdditionalImageFields image_util.FsSlice `json:"additionalImageFields,omitempty" yaml:"additionalImageFields,omitempty"`
+	// context logs each detailed result
+	context *fn.Context
+	// resultCount logs the total count image change
+	resultCount int
+}
+
+// Config transforms the data from ConfigMap to SetImage struct
+func (imagetrans *SetImage) Config() error {
```

Comment:
Can you verify the previous comment is resolved? See the second example. https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/904#discussion_r953264086

Reply:
I am not sure which one I missed here, I have given every variable name ` to show the exact name. And I have changed T to t.

Reply:
Oh I see!! you mean change the whole name to `t`.